src/utils/reporting_workflow.js

```diff
@@ -45,13 +50,130 @@ const IG_REGISTRY = {
   [BASE_REPORTING_BUNDLE_PROFILE]: {} // base action definitions, see MEDMORPH-35
 };
 
-function initializeContext(planDefinition) {
-  // TODO -- see MEDMORPH-36
+/**
+ * Retrieve data from the EHR
+ *
+ * @param {string} uri - the search query for the data desired
+ * @returns axios promise with data
+ */
+async function readFromEHR(uri) {
+  const url = process.env.EHR;
+  const token = await connectToServer(url);
+  const headers = { Authorization: `Bearer ${token}` };
+  return axios
+    .get(`${url}/${uri}`, { headers: headers })
+    .then(response => response.data)
+    .catch(err => console.error(err));
+}
+
+/**
+ * Post the reporting bundle to the endpoint designated in the PlanDefinition
+ *
+ * @param {PlanDefinition} planDefinition - the current PlanDefinition from the context
+ * @param {Bundle} bundle - the reporting Bundle to submit
+ * @returns axios promise with data
+ */
+async function submitBundle(planDefinition, bundle) {
+  const receiverAddress = planDefinition.extension.find(e => e.url === RECEIVER_ADDRESS_EXT);
+  const endpointRef = receiverAddress.valueReference.reference;
+  const endpointId = endpointRef.split('/')[1];
+  const endpoint = db.select('endpoints', e => e.id === endpointId);
+  const url = endpoint.address;
+  // Not sure how this will work since the url comes from endpoint.address....
+  const token = await connectToServer(url);
+  const headers = { Authorization: `Bearer ${token}` };
+  return axios
+    .post(url, bundle, { headers: headers })
+    .then(response => response.data)
+    .catch(err => console.error(err));
+}
+
+/**
+ * Helper function to call specific operations on the data trust service server.
+ *
+ * @param {string} operation = 'deidentify' | 'anonymize' | 'pseudonymize'
+ * @param {Bundle} bundle - the bundle to perform operation on
+ * @returns axios promise with data
+ */
+function dataTrustOperation(operation, bundle) {
+  return axios
+    .post(`${process.env.DATA_TRUST_SERVICE}/Bundle/$${operation}`, bundle)
+    .then(response => response.data)
+    .catch(err => console.error(err));
+}
+
+/**
+ * Function to use the PlanDefinition and triggering resource to create a report
+ *
+ * @param {PlanDefinition} planDef - the PlanDefinition resource
+ * @param {*} resource - the resource which triggered the notification
+ */
+async function initializeReportingWorkflow(planDef, resource = null) {
+  // TODO: MEDMORPH-49 to make sure the resource is always included
+  if (!resource) return;
+
+  let patient = null;
+  if (resource.patient) patient = await getReferencedResource(resource.patient.reference);
```

Comment:
doesn't this take 2 arguments?  The patient reference will be accepted as the `url` argument